analyses/DESeq_analysis/run-DESeq-analysis.R

```diff
@@ -0,0 +1,195 @@
+}
+
+#Create an empty df to populate with rbind of all tumor Kids_First_Biospecimen_ID and cancer_group by cohort
+#Create DF that list all Kids_First_Biospecimen_IDs by Cohort - Cancer groups
+sample_type_df_tumor_cohort = data.frame()
+for(I in 1:length(Cancer_Histology_COHORT))
+{
+  Cancer_Histology_COHORT_cohort =  strsplit(Cancer_Histology_COHORT[I],split="_")[[1]][1]
+  Cancer_Histology_COHORT_cancer_group =  strsplit(Cancer_Histology_COHORT[I],split="_")[[1]][2]
+  sample_type_df_tumor_cohort = rbind(sample_type_df_tumor_cohort,data.frame(Case_ID = hist.filtered$Kids_First_Biospecimen_ID[which(hist.filtered$cancer_group == Cancer_Histology_COHORT_cancer_group & hist.filtered$cohort == Cancer_Histology_COHORT_cohort)],Type=Cancer_Histology_COHORT[I]))
+}
+
+#Combine the rows from the normal and tumor sample df
+sample_type_df = rbind(sample_type_df_tumor,sample_type_df_tumor_cohort,sample_type_df_normal)
+
+#Filter one more to ensure the rownames in the countsdata file match the sample dataframe for DEG just created
+countData_filtered_DEG = countData_filtered[,which(colnames(countData_filtered) %in% sample_type_df$Case_ID)]
+
+#Filter one more to ensure the rownames in the countsdata file match the sample dataframe for DEG just created
+sample_type_df_filtered = unique(sample_type_df[which(sample_type_df$Case_ID %in% colnames(countData_filtered_DEG)),])
+
+#Define All cancer groups (Combined and cohort-specific) in the histology list
+histology_filtered = unique(sample_type_df_filtered$Type[-grep("GTEX",sample_type_df_filtered$Case_ID)])
+
+#Define All GTEx groups as normal in the GTEX_filtered list
+GTEX_filtered = unique(sample_type_df_filtered$Type[grep("GTEX",sample_type_df_filtered$Case_ID)])
+
+#Assign cmparison
+ I = as.numeric(HIST_index)   #assign first argument to Histology index
+ J = as.numeric(GTEX_index)   #assign second argument to GTEx index
+
+
+    
+#Subset countData_filtered_DEG dataframe for only the histology group and GTEx group being compared
+    countData_filtered_DEG.hist = data.matrix(countData_filtered_DEG[,which(colnames(countData_filtered_DEG) %in% sample_type_df_filtered$Case_ID[which(sample_type_df_filtered$Type %in% c(histology_filtered[I],GTEX_filtered[J]))])])
+    
+#Subset sample type dataframe for only the histology group and GTEx group being compared
+    sample_type_df_filtered.hist = sample_type_df_filtered[which(sample_type_df_filtered$Type %in% c(histology_filtered[I],GTEX_filtered[J])),]
+    
+#Run DESeq2  
+    sub.deseqdataset <- DESeqDataSetFromMatrix(countData=round(countData_filtered_DEG.hist),
+                                               colData=sample_type_df_filtered.hist,
+                                               design= ~ Type)
+    
+    
+    sub.deseqdataset$Type <- factor(sub.deseqdataset$Type, levels=c(GTEX_filtered[J], histology_filtered[I]))
```

Comment:
It would be helpful to note somewhere in the module or result table that the log fold change in the result is GTEX / cancer_group.